### PR TITLE
[O2B-1115] Implement runs per data pass API

### PR DESCRIPTION
**Pull request by xsalonx**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [x] issue has "Fix version" assigned
- [x] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for users:
- NA

Notable changes for developers:
- Add filtering of runs by data pass id
- Fix `QueryBuilder#whereAssociation` through-junction-table associations

Changes made to the database:
- NA

### CODE REVIEW

**Comment by xsalonx** [comment]
To merged after #1331

File: lib/usecases/run/GetAllRunsUseCase.js [hunk 1]
```diff
@@ -233,6 +235,15 @@ class GetAllRunsUseCase {
                 });
             }
 
+            if (dataPassIds) {
+                const runNumbers = (await RunRepository.findAll(dataSource.createQueryBuilder()
+                    .set('attributes', ['runNumber'])
+                    .set('raw', true)
+                    .setModel('Run')
+                    .whereAssociation('dataPass', 'id').oneOf(...dataPassIds))).map(({ runNumber }) => runNumber);
+                filteringQueryBuilder.where('runNumber').oneOf(...runNumbers);
```
Comment (xsalonx) [inline lib/usecases/run/GetAllRunsUseCase.js:244]: direct use of `whereAssociation(...)` with `filteringQueryBuilder` doesn't work

File: lib/usecases/run/GetAllRunsUseCase.js [hunk 2]
```diff
@@ -233,6 +235,15 @@ class GetAllRunsUseCase {
                 });
             }
 
+            if (dataPassIds) {
+                const runNumbers = (await RunRepository.findAll(dataSource.createQueryBuilder()
+                    .set('attributes', ['runNumber'])
+                    .set('raw', true)
+                    .setModel('Run')
+                    .whereAssociation('dataPass', 'id').oneOf(...dataPassIds))).map(({ runNumber }) => runNumber);
```
Comment (martinboulais) [inline lib/usecases/run/GetAllRunsUseCase.js:243]: TBH here, you should simply use raw query object, as it is done for tags, don't bother with query builder

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: lib/usecases/run/GetAllRunsUseCase.js (hunk 2 above)
Comment (xsalonx) [reply lib/usecases/run/GetAllRunsUseCase.js]: applied

**Review by martinboulais — APPROVED** [review]

**Pull request merged**